src/pgsql.c

```diff
@@ -137,6 +172,20 @@ kore_pgsql_query_params(struct kore_pgsql *pgsql, struct http_request *req,
 	return (KORE_RESULT_OK);
 }
 
+int
+kore_pgsql_query_sync(struct kore_pgsql *pgsql, const char *query) 
+{
+	pgsql->result = PQexec(pgsql->conn->db, query);
+	/* Success only for SELECTS or COMMANDS */
+        if ((PQresultStatus(pgsql->result) != PGRES_TUPLES_OK) &&
+		(PQresultStatus(pgsql->result) != PGRES_COMMAND_OK)) {
+		pgsql->state = KORE_PGSQL_STATE_ERROR;
```

Comment:
If this fails, set the pgsql->error as well in the the form of:

pgsql->error = kore_strdup(PQerrorMessage(pgsql->conn->db));

So it is available for the caller.